src/minimalEntryPoint.js

```diff
@@ -9,19 +9,22 @@ export function minimalEntryPoint({
   reducers,
   router: Router,
   initialState,
+  middleware = [],
   rootSelector = '#container',
 }) {
   const rootNode = document.querySelector(rootSelector);
   const supportsHistory = 'pushState' in window.history;
-  let storeCreator = createStore;
+
   if (process.env.NODE_ENV !== 'production') {
     try {
       // eslint-disable-next-line global-require, import/no-extraneous-dependencies
-      storeCreator = applyMiddleware(require('redux-logger').logger)(storeCreator);
+      middleware.unshift(require('redux-logger').logger);
```

Comment:
I don't remember all the details here, but I remember doing this nutty storeCreator thing to avoid pulling in applyMiddleware unless it was needed to keep bundle size small. I guess with this approach, we can't avoid that middleware tax.

Reply:
Alternatively, the argument could be "createStore" defaulting to createStore, then I think it gets around the problem? Assuming you can applyMiddleware twice.

Reply:
That is a good point, I would imagine we could just check if any middleware exists in the array and apply it in that block and get the expected optimization?

```
const storeCreator = middleware.length ? applyMiddleware(...middleware)(createStore) : createStore;
```